pkg/intermediate/aggregate.go

```diff
@@ -0,0 +1,218 @@
+package intermediate
+
+import (
+	"encoding/binary"
+	"fmt"
+	"net"
+	"sync"
+
+	"github.com/vmware/go-ipfix/pkg/entities"
+	"github.com/vmware/go-ipfix/pkg/registry"
+)
+
+type aggregation struct {
+	// tupleRecordMap maps each connection with its records
+	tupleRecordMap map[Tuple][]entities.Record
+	// tupleRecordLock allows multiple readers or one writer at the same time
+	tupleRecordLock sync.RWMutex
+	// workerPool is for exchanging worker channels to process message
+	workerPool chan chan *entities.Message
+	// messageChan is the channel to receive message
+	messageChan chan *entities.Message
+	// workerNum is the number of workers to process message
+	workerNum int
+	// workerList is the list of workers
+	workerList []*worker
+}
+
+type Tuple struct {
+	// SourceAddress is the encoded format of sourceIPv4Address
+	SourceAddress uint32
```

Comment:
`net.IP` is a slice type, which cannot be used as struct field as map key because keyType of map should be [comparable](https://golang.org/ref/spec#Comparison_operators). reference: https://blog.golang.org/maps. 
Changed to array type which will work for both IPv4 and IPv6

Reply:
Let's use string type for IPs?

Reply:
Great idea. Thanks!